### PR TITLE
Booking/return

### PR DESCRIPTION
**Pull request by sbambur**
- Completed booking/return book logic
- Added images .png
- Implemented custom hooks for timeout and debounce
- Added validation for date input (you can't choose a date earlier than tomorrow)
- Refactoring of some components and functions

### CODE REVIEW

File: src/components/BookItem/style.css [hunk 1]
```diff
@@ -37,6 +46,29 @@ a:not(.locked) .bookCard:hover {
...
+  transition: opacity 0.2s ease-out;
+  justify-content: flex-end;
+}
+
+.return_date .return_button .button {
+  z-index: 15;
+  margin-bottom: 40px;
+  font-size: 20px;
+  line-height: 46px;
+  line-height: 24px;
```
Comment (dariausatova) [inline src/components/BookItem/style.css:63]: line-height 2 times

File: src/components/utils/bookingCounter.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+export const getBookingBook = (quantity = 0) => {
+  const string = quantity.toString();
+  const lastChar = string.charAt(string.length - 1);
+
+  switch (true) {
+    case lastChar === '1' && !(quantity === 11):
+      return `${quantity} книга`;
+    case lastChar === '2' && !(quantity === 12):
+    case lastChar === '3' && !(quantity === 13):
+    case lastChar === '4' && !(quantity === 14):
```
Comment (dariausatova) [inline src/components/utils/bookingCounter.ts:10]: and there won't be more than 4 books?

File: src/hooks/useDebounce.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+import useTimeout from 'hooks/useTimeout';
+
+export default function useDebounce<T extends Function>(
+  callback: T,
+  delay: number,
+  dependencies: any
+) {
+  const { reset, clear } = useTimeout(callback, delay);
+  useEffect(reset, [...dependencies, reset]);
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (dariausatova) [inline src/hooks/useDebounce.ts:11]: remove this line

File: src/hooks/useTimeout.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import { useCallback, useEffect, useRef } from 'react';
+
+export default function useTimeout<T extends Function>(callback: T, delay: number) {
```
Comment (dariausatova) [inline src/hooks/useTimeout.ts:3]: es6

File: src/public/index.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export const ICONS = {
```
Comment (dariausatova) [inline src/public/index.ts:1]: can't I use the library for icons?

File: src/components/utils/bookingCounter.ts (hunk 2 above)
Comment (sbambur) [reply src/components/utils/bookingCounter.ts:10]: For this case, the default option is used

File: src/public/index.ts (hunk 5 above)
Comment (sbambur) [reply src/public/index.ts:1]: The icons are taken from a template, what is the library for?

Comment (dariausatova) [reply src/public/index.ts:1]: these are standard icons and you can connect any library with them and not use them from the layout

File: src/components/Book/ReturnBookModal.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import { FC, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import moment from 'moment';
+import { useReturnDate } from 'hooks/useReturnDate';
+import { useInterval } from 'hooks/useInterval';
+import { Modal } from 'react-bootstrap';
+import useStore from 'hooks/useStore';
+
+interface ReturnBookModalProps {
```
Comment (dariausatova) [inline src/components/Book/ReturnBookModal.tsx:9]: please take out all interfaces separately

**Pull request merged**